app/controllers/physical_storage_controller.rb

```diff
@@ -4,12 +4,35 @@ class PhysicalStorageController < ApplicationController
   include Mixins::GenericSessionMixin
   include Mixins::MoreShowActions
   include Mixins::BreadcrumbsMixin
+  include Mixins::GenericButtonMixin
 
   before_action :check_privileges
   before_action :session_data
   after_action :cleanup_action
   after_action :set_session_data
 
+  def specific_buttons(pressed)
+    case pressed
+    when 'physical_storage_new'
+      javascript_redirect(:action => 'new')
+    else
+      return false
+    end
+    true
+  end
+
+  def new
+    assert_privileges("physical_storage_new")
```

Comment:
You only need this call if the `${controller_name}_#{controller_action}` would not get you the feature name, but in this case you're covered.